### PR TITLE
Dump session keys

### PR DESCRIPTION
**Pull request by mpetrun5**
Log session keys when created and exchanged.

Related: #16

### CODE REVIEW

File: src/constants.ts [hunk 1]
```diff
@@ -1,3 +1,6 @@
 export const NOISE_MSG_MAX_LENGTH_BYTES = 65535;
 export const NOISE_MSG_MAX_LENGTH_BYTES_WITHOUT_TAG = NOISE_MSG_MAX_LENGTH_BYTES - 16;
 
+export const DUMP_SESSION_KEYS = false;
```
Comment (mpetrunic) [inline src/constants.ts:4]:
```suggestion
export const DUMP_SESSION_KEYS = process.env.NOISE_DUMP_KEYS;
```

Comment (mpetrun5) [reply src/constants.ts:4]: You already need DEBUG=libp2p:session as an enviroment variable to show session keys. This  const made sense so you won't accidentally leave an enviroment variable active and show them. So in my opinion either we don't need this variable or leave as a constant.

Comment (mpetrunic) [reply src/constants.ts:4]: oh don't do that, libp2p:session is meant for logs called js-libp2p-session. Usually people will log libp2p stuff with DEBUG=libp2p:* which would log keys by accident and we should avoid that

Comment (mpetrun5) [reply src/constants.ts:4]: Definitely should change DEBUG to something more specific to this, eg. noise:session-keys?

Comment (mpetrunic) [reply src/constants.ts:4]: it should stay libp2p:noise, but logging of keys should depend on additional env variable

File: src/logger.ts [hunk 2]
```diff
@@ -1,2 +1,51 @@
 import debug from "debug";
-export const logger = debug('libp2p:noise');
+import {DUMP_SESSION_KEYS} from './constants';
+import { KeyPair } from "./@types/libp2p";
+import { NoiseSession, SymmetricState } from "./@types/handshake";
+
+let keyLogger;
+if(DUMP_SESSION_KEYS){
+  keyLogger = debug('libp2p:session')
```
Comment (mpetrunic) [inline src/logger.ts:8]: Use existing logger instead of debug('libp2p:session')

File: src/handshake-xx.ts [hunk 3]
```diff
@@ -50,11 +58,13 @@ export class XXHandshake implements IHandshake {
...
   public async propose(): Promise<void> {
+    logLocalStaticKeys(this.session.hs.s)
     if (this.isInitiator) {
       logger("Stage 0 - Initiator starting to send first message.");
       const messageBuffer = this.xx.sendMessage(this.session, Buffer.alloc(0));
       this.connection.writeLP(encode0(messageBuffer));
       logger("Stage 0 - Initiator finished sending first message.");
+      logLocalEphemeralKeys(this.session.hs.e)
     } else {
       logger("Stage 0 - Responder waiting to receive first message...");
```
Comment (morrigan) [inline src/handshake-xx.ts:69]: I think you are missing `logLocalStaticKeys` for the responder too

**Pull request merged**